src/bindings/c/ov/include/c_api/ov_c_api.h

```diff
@@ -0,0 +1,1122 @@
+    NUM_STREAMS,                        //  Read-write property to set/get the number of executor logical partitions
+    AFFINITY,                           //  Read-write property to set/get the name for setting CPU affinity per thread option.
+    INFERENCE_NUM_THREADS,              //  Read-write property to set/get the maximum number of threads that can be used for inference tasks
+    PERFORMANCE_HINT,                   //  Read-write property, it is high-level OpenVINO Performance Hints unlike low-level properties that are individual (per-device),
+                                        //  the hints are something that every device accepts and turns into device-specific settings
+                                        //  detail see ov_performance_mode_e to get its hint's key name
+    NETWORK_NAME,                       //  Read-only property to get a name of name of a model
+    INFERENCE_PRECISION_HINT,           //  Read-write property to set the hint for device to use specified precision for inference
+    OPTIMAL_BATCH_SIZE,                 //  Read-only property to query information optimal batch size for the given device and the network
+    MAX_BATCH_SIZE,                     //  Read-only property to get maximum batch size which does not cause performance degradation due to memory swap impact.
+    PERFORMANCE_HINT_NUM_REQUESTS,      //  (Optional) property that backs the Performance Hints
+                                        //  by giving additional information on how many inference requests the application will be keeping in flight
+                                        //  usually this value comes from the actual use-case (e.g. number of video-cameras, or other sources of inputs)
+} ov_property_key_e;
+
+typedef union {
+    uint32_t value_u;
+    char value_s[320];
+    ov_performance_mode_e value_performance_mode;
+}ov_property_value;
+
+typedef struct ov_property{
+    ov_property_key_e key;
+    ov_property_value value;
+    struct ov_property* next;
+}ov_property_t;
+
+/**
+ * @brief Initialize a partial shape.
+ * @param ov_status_e a status code.
+ */
+OPENVINO_C_API(ov_status_e) ov_partial_shape_init(ov_partial_shape_t* partial_shape, const char* str);
+
+/**
+ * @brief Parse the partial shape to readable string.
+ * @param ov_status_e a status code.
+ */
+OPENVINO_C_API(const char*) ov_partial_shape_parse(ov_partial_shape_t* partial_shape);
+
+/**
+ * @brief Release partial shape.
+ * @param ov_status_e a status code.
+ */
+OPENVINO_C_API(void) ov_partial_shape_free(ov_partial_shape_t* partial_shape);
+
+/**
+ * @brief Covert partial shape to static shape.
+ * @param ov_status_e a status code, return OK if successful
+ */
+OPENVINO_C_API(ov_status_e) ov_partial_shape_to_shape(ov_partial_shape_t* partial_shape, ov_shape_t* shape);
+
+/**
+ * @brief Print the error info.
+ * @param ov_status_e a status code.
+ */
+OPENVINO_C_API(const char*) ov_get_error_info(ov_status_e status);
```

Comment:
should we free memory after? or this is a reference to an internal constant

Reply:
it will be freed by ov_free() API.